docs-sphinx/user-guide/how-to-use-header-only-layoutbuilder.md

```diff
@@ -0,0 +1,260 @@
+---
+jupytext:
+  text_representation:
+    extension: .md
+    format_name: myst
+    format_version: 0.13
+    jupytext_version: 1.14.1
+kernelspec:
+  display_name: C++14
+  language: C++14
+  name: xcpp14
+---
+
+How to use the header-only LayoutBuilder in C++
+===============================================
+
+What is header-only Layout Builder?
+-----------------------------------
+
+The header-only Layout Builder consists of a set of compile-time, templated, static C++ classes, implemented entirely in header file which can be dropped into any external project, and easily separable from the rest of the Awkward C++ codebase.
+
+The Layout Builder namespace consists of the following 14 types of Layout Builders:
+
+* `Numpy`
+* `ListOffset`
+* `List`
+* `Empty`
+* `EmptyRecord`
+* `Record`
+* `Tuple`
+* `Regular`
+* `Indexed`
+* `IndexedOption`
+* `Unmasked`
+* `ByteMasked`
+* `BitMasked`
+* `Union`
+
+All Builders except `Numpy` and `Empty` can take any other Builder as template parameters.
+These Builders are sufficient to build every type of Awkward Array.
+
+Why header-only Layout Builder?
+-------------------------------
+
+The users can directly include `LayoutBuilder.h` in their compilation, rather than linking against platform-specific libraries or worrying about native dependencies. This makes the integration of Awkward Arrays into other projects easier and more portable.
+
+The code is minimal, it does not include all of the code needed to use Awkward Arrays in Python, nor does it include references to Python or pybind11. The C++ users can use it to make arrays and then copy them to Python without any specialised data types - only raw buffers, strings, and integers.
+
+How to use Layout Builders?
+-----------------------------
+
+The following cpp-headers which reside in this [directory](https://github.com/scikit-hep/awkward/tree/main/src/awkward/_v2/cpp-headers/awkward) are needed to use Layout Builders.
+
+1. BuilderOptions.h
+2. GrowableBuffer.h
+3. LayoutBuilder.h
+4. utils.h
+
+Three phases of using Layout Builder
+------------------------------------
+
+1. <b>Constructing a Layout Builder:</b> from variadic templates. (It is implicit template instantiation).
+2. <b>Filling the Layout Builder:</b> while repeatedly walking over the raw pointers within the LayoutBuilder.
+3. <b>Taking the data out to user allocated buffers:</b> Then user can pass them to Python if they want.
+
+Example
+-------
+
+Below is an example for creating RecordArray with NumpyArray and ListOffsetArray as fields.
+
+Type alias can be used for each builder class.
+
```

Comment:
Please, add some clarification why there is only one header file included. You've mentioned a few before.